tests/utils.py

```diff
@@ -52,6 +52,68 @@ def execute_pysvzerod(testfile, mode):
     return result, config
 
 
+def run_with_reference(
+        ref,
+        test_config
+        ):
+
+
+    res = pysvzerod.simulate(test_config)
+
+    if res.shape[1] == 6:
+        # we have a result with fields [name, time, p_in, p_out, q_in, q_out]
+        for field in ["pressure_in", "pressure_out", "flow_in", "flow_out"]:
+            if "pressure" in field:
+                assert np.isclose(res[field].to_numpy().all(), ref[field].to_numpy().all(), rtol=RTOL_PRES)
+            elif "flow" in field:
+                assert np.isclose(res[field].to_numpy().all(), ref[field].to_numpy().all(), rtol=RTOL_FLOW)
+    else:
+        # we have a result with fields [name, time, y] and the result must be compared based on the name field. name is of format [flow:vessel:outlet]
```

Comment:
Why is this being compared only with the average instead of the full solution as you are doing above when the result has fields [name, time, p_in, p_out, q_in, q_out]?

Reply:
This is being compared with the average because there appears to be some discrepancy in the computation between operating systems. The test will not pass on either Mac or ubuntu (depending on which OS was used to compute the solution) if the full solution is compared. This will be fixed to compare to the full solution.